### PR TITLE
Fix build warnings in CA snippets

### PR DESCRIPTION
**Pull request by gewarren**

### CODE REVIEW

**Comment by gewarren** [comment]
@GrabYourPitchforks Could you please review this since it involves removing code/verbiage around CAS?

File: docs/fundamentals/code-analysis/quality-rules/snippets/csharp/all-rules/.editorconfig [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+dotnet_diagnostic.CS0659.severity = suggestion
+
+# CAXXXX analyzers
+dotnet_diagnostic.CA1822.severity = none
+
+# CS0108: Member hides inherited member; missing new keyword
+dotnet_diagnostic.CS0108.severity = suggestion
+
+[*.xml]
+indent_size = 2
```
Comment (Youssef1313) [inline docs/fundamentals/code-analysis/quality-rules/snippets/csharp/all-rules/.editorconfig:18]: There is already an .editorconfig in repo root. I'm not sure why a new one is needed?

**Review by BillWagner — APPROVED** [review]
This LGTM @gewarren 

You can :shipit: once @GrabYourPitchforks weighs in.

**Comment by GrabYourPitchforks** [comment]
I have no problem with the verbiage change. However, it is important to note that CAS is well and truly dead, even on Full Framework. Rules like CA2109 no longer have meaning and IMO should be deleted from any static analysis package we provide.

**Pull request merged**